### ISSUE TITLE
[HOLD for payment 2023-05-05] [$2000] Uploading gif image seems to fluctuate the image while previewing it immediately after uploading

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16528]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open web chrome and open any chat and send a message
2. Now click on profile icon and upload a gif image. Crop the image to full.
3. From the chat, click on your profile picture and preview it.
4. First Bug : Notice that the image fluctuates
5. Second Bug : Also, sometimes an empty attachment also gets displayed (This is not easily reproducible) For this,  upload a large jpg/png file without cropping and try to preview the image immediately. It displays an empty attachment. - _Not able to reproduce for me_ 

## Expected Result:
The image shouldn’t fluctuate while trying to preview the image

## Actual Result:
The image fluctuates while trying to preview the image

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.89-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/227808677-8d6ac413-894e-496f-a8f4-58c4cb76a459.mp4

https://user-images.githubusercontent.com/43996225/227808679-84e12dbd-fd2b-47d1-9de5-1815f8b49075.mp4

https://user-images.githubusercontent.com/43996225/227808680-c12a3b70-07ed-4a2d-ba0d-0a64709e965f.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @priya-zha
**Slack conversation:** 
https://expensify.slack.com/archives/C049HHMV9SM/p1679819714964549
[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01afe7983a04d30b23</li>
        <li>Upwork Job ID: 1643073372561469440</li>
        <li>Last Price Increase: 2023-04-25</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @stephanieelliott (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by priya-zha** [comment]
@marcaaron It happens with normal images too. Can you please try to view any profile picture of the user. It seems to fluctuate with normal images too, not only with gifs.

**Comment by stephanieelliott** [comment]
Hm, I'm not seeing that behavior occur on static images, like .jpg. For me, it's only happening with .gifs. 

Agree with @marcaaron that we probably shouldn't support .gifs in that field, posted in Slack to discuss: https://expensify.slack.com/archives/C01SKUP7QR0/p1680106537787609

**Comment by stephanieelliott** [comment]
OK it sounds like we had previously agreed to allow a GIF and convert it to JPG (context). So I suppose this does need to be fixed.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01afe7983a04d30b23

**Comment by MelvinBot** [comment]
Current assignee @stephanieelliott is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @eVoloshchak (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @danieldoglas (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by tienifr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The image fluctuates while trying to preview the image

### What is the root cause of that problem?
The issue is in this line where we're setting the width and height of the `Pressable` that wraps the `Image` to `100%` if the image width and height are 0 (a.k.a the image is still loading).

When the images finish loading, the image shows as full size (100%) while the image is very small (less than the full size), so it immediately resizes to its actual small size, causing the flicker.

### What changes do you think we should make in order to solve the problem?

We should not set width and height of the image container if the image is still loading. This can be done by `return undefined` in here https://github.com/Expensify/App/blob/3c411711f9c641338694a85c0ebfd9a9ec97a9e2/src/styles/StyleUtils.js#L93 if the image is loading.

We can pass the loading state in by using the `this.state.isLoading` of the `ImageView` component.

### What alternative solutions did you explore? (Optional)
NA

**Comment by danieldoglas** [comment]
@tienifr don't know if you saw [this comment](https://github.com/Expensify/App/issues/16528#issuecomment-1495238430), but the expectation here is to convert the GIF to a JPG. I don't think your proposal is doing that.

**Comment by tienifr** [comment]
> convert the GIF to a JPG

@danieldoglas this is an existing feature on staging, the Avatar upload is already doing this for uploaded GIFs.
I think what @stephanieelliott means here is that since uploading GIFs is a valid operation, this bug of flickering images should be fixed, my proposal addresses that.

Also just to clarify the issue occurs even for regular png/jpg images (not just GIFs), the key here is to use an image with width and height smaller than that of the desktop web container width and height, you can try with an image of just 100px by 100px.

**Comment by PrashantMangukiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Small Image flicker on web and desktop when opened via attachment modal, this is applicable for gif or regular image.

### What is the root cause of that problem?
Within `ImageView/index.js` we are using `getZoomSizingStyle()` function  at line 277
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/components/ImageView/index.js#L277

This `getZoomSizingStyle()` function is as under. 
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/styles/StyleUtils.js#L201-L207

When component load at that time `isZoomed` is false, so it will return height and width 100% as shown on line 204, 205 so  image show full size until correct dimension received. This is the root cause.

### What changes do you think we should make in order to solve the problem?
Within `StyleUtils.js` update `getZoomSizingStyle()` function as shown below. 
i.e. When image dimension not decided return undefined.
```javascript
if (imgWidth === 0 || imgHeight === 0) {
   return {
       //height: isZoomed ? '250%' : '100%',    // Old Code
       height: isZoomed ? '250%' : undefined,   // Updated Code
       //width: isZoomed ? '250%' : '100%',       // Old Code
       width: isZoomed ? '250%' : undefined,   // Updated Code
   };
}
```

### What alternative solutions did you explore? (Optional)
Just return undefined object as shown below:
```javascript
  if (imgWidth === 0 || imgHeight === 0) {
      return undefined;
  }
```

<details>
<summary> Result</summary>

https://user-images.githubusercontent.com/7823358/229275760-a7805bfa-3204-4799-8184-962b4844242a.mp4

</details>

**Comment by eVoloshchak** [comment]
Reviewing proposals above tomorrow

**Comment by eVoloshchak** [comment]
I like @tienifr's [proposal](https://github.com/Expensify/App/issues/16528#issuecomment-1495303707), `Hide image until finished loading to prevent showing preview with wrong dimensions` is a reasonable approach.


> Second Bug : Also, sometimes an empty attachment also gets displayed (This is not easily reproducible) For this, upload a large jpg/png file without cropping and try to preview the image immediately. It displays an empty attachment. - Not able to reproduce for me

I've managed to reproduce this (it's easier if you have a slow connection):

1. Click on profile icon and upload a gif image (can be any size)
2. From the chat, click on your profile picture and preview it (you have to do this immediately, so using slow 3G preset helps)

@tienifr, are you able to reproduce this?

**Comment by tienifr** [comment]
> you have to do this immediately, so using slow 3G preset helps

@eVoloshchak thanks for the explanation on the reproduction steps, I can reproduce it now, here's my proposal for that issue (it's a very different issue with different root cause)

## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Sometimes an empty attachment gets displayed when uploading a large avatar in a slow network connection, and then view the avatar immediately in the chat detail page.

### What is the root cause of that problem?
The root cause is here https://github.com/Expensify/App/blob/64f4a7814e87bd7c4d085776abf5cb8a67b9e8bc/src/components/AttachmentView.js#L84 where we're checking if the attachment is an image or not.

When we upload the image and view it immediately, the image URL is a blob `blob:http://localhost:8080/74a3cc86-0cc8-4e09-a9cb-509e23979e90` so it doesn't have the image extension for the image check. The `file` here is also empty because it will only be passed in the attachment view if you just uploaded it in the avatar upload page, here we're accessing the avatar in a completely different component (chat details).

So the component is unable to check that the source passed in is an image, thus it displays the paper clip icon as a fallback.

### What changes do you think we should make in order to solve the problem?

There're certain scenarios where we know exactly that the `AttachmentView` will display an image, for example here https://github.com/Expensify/App/blob/6ccfc4b4165f4375757c65e25bf0f59b72602063/src/pages/DetailsPage.js#L122 where we display the avatar.

We can just pass an `isImage` prop to the `AttachmentModal` on that page (since it's displaying an avatar, we know it is an image), then pass `isImage` down to the `AttachmentView`. 

And update the condition here https://github.com/Expensify/App/blob/64f4a7814e87bd7c4d085776abf5cb8a67b9e8bc/src/components/AttachmentView.js#L84 so that if `isImage` is true, it will render an image, otherwise it will depend on the checking of the `source` and `file` as usual

### What alternative solutions did you explore? (Optional)
NA

**Comment by MelvinBot** [comment]
@danieldoglas @eVoloshchak @stephanieelliott this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by hellohublot** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Uploading gif image seems to fluctuate the image while previewing it immediately after uploading

### What is the root cause of that problem?

Why does a big picture flash
- Because `ImageView.constructor.this.state.isLoading = false`, and `ImageView.onLoadStart.this.setState({ isLoading: true })`, the `setState` requires execution time, so the styles are `styles.h100` and `styles.w100` at the `beginning`, So we'll see a big picture flash

### What changes do you think we should make in order to solve the problem?

1. We can modify `ImageView.constructor.this.state.isLoading = true`
2. In order to support gif conversion, we can insert `cropOrRotateImage` before `AvatarWithImagePicker.showAvatarCropModal.this.setState`


### What alternative solutions did you explore? (Optional)
Not Yet

**Comment by MelvinBot** [comment]
@danieldoglas, @eVoloshchak, @stephanieelliott Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@danieldoglas, @eVoloshchak, @stephanieelliott Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by eVoloshchak** [comment]
@tienifr, awesome!
I think the first alternative solution you've [proposed](https://github.com/Expensify/App/issues/16528#issuecomment-1499949605) is a more universal approach, we won't have to rely on all the necessary components passing `isImage` prop in the future

> Another solution we can use is to pass the file name (aside from the avatar which will be the blob URL) to the optimistic data when we upload the avatar, then pass the avatar file name to the AttachmentModal and use it as the default file state of the AttachmentModal. This will also work and quite straight-forward.

Could you please post an updated proposal that contains the fix for both bugs?

**Comment by eVoloshchak** [comment]
> We can modify ImageView.constructor.this.state.isLoading = true

@hellohublot [proposal](https://github.com/Expensify/App/issues/16528#issuecomment-1501105224) looks good for resolving the first bug. This seems to be the actual root cause, I don't think there is a case where we'd want isLoading to be true when the component is first mounted

> In order to support gif conversion, we can insert cropOrRotateImage before AvatarWithImagePicker.showAvatarCropModal.this.setState

I don't think there is a problem with gif conversion, they already get converted to static images. Is there another issue with GIFs?

> In order to solve the problem that AttachmentModal.state.source is not refreshed after uploading, we can update AttachmentModal.state.source in AttachmentModal.componentDidUpdate by AttachmentModal.props.source

Could you elaborate a bit more on this?

**Comment by tienifr** [comment]
@eVoloshchak Thanks for the review! here's my combined proposal:

## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There're 2 issues here:
1. The image fluctuates while trying to preview the image
2. Sometimes an empty attachment gets displayed when uploading a large avatar in a slow network connection, and then view the avatar immediately in the chat detail page.

### What is the root cause of that problem?
For **1.**
The issue is in [this line](https://github.com/Expensify/App/blob/b96c5167e7ac1f9f4ebfdd61311433c421871695/src/styles/StyleUtils.js#L204) where we're setting the width and height of the `Pressable` that wraps the `Image` to `100%` if the image width and height are 0 (a.k.a the image is still loading).

When the images finish loading, the image shows as full size (100%) while the image is very small (less than the full size), so it immediately resizes to its actual small size, causing the flicker.

For **2.**
The root cause is here https://github.com/Expensify/App/blob/64f4a7814e87bd7c4d085776abf5cb8a67b9e8bc/src/components/AttachmentView.js#L84 where we're checking if the attachment is an image or not.

When we upload the image and view it immediately, the image URL is a blob `blob:http://localhost:8080/74a3cc86-0cc8-4e09-a9cb-509e23979e90` so it doesn't have the image extension for the image check. The `file` here is also empty because it will only be passed in the attachment view if you just uploaded it in the avatar upload page, here we're accessing the avatar in a completely different component (chat details).

So the component is unable to check that the source passed in is an image, thus it displays the paper clip icon as a fallback.

### What changes do you think we should make in order to solve the problem?

For **1.**
We should not set width and height of the image container if the image is still loading. This can be done by `return undefined` in here https://github.com/Expensify/App/blob/3c411711f9c641338694a85c0ebfd9a9ec97a9e2/src/styles/StyleUtils.js#L93 if the image is loading.

We can pass the loading state in by using the `this.state.isLoading` of the `ImageView` component.

Please note we're already trying to do the same here https://github.com/Expensify/App/blob/6bfc8bbe641645174c7c01d3ac664535730442ab/src/components/ImageView/index.js#L235 with the `Image` component, but it doesn't work well in this case since the Image and its container renders in different times so there is race condition. That logic should be applied to the container of the image instead, as suggested above. We can safely remove this check for web if the container check above is applied (however that check in case of mWeb [here](https://github.com/Expensify/App/blob/6bfc8bbe641645174c7c01d3ac664535730442ab/src/components/ImageView/index.js#L235) should still be kept since there's no `Pressable` container on mWeb).

For **2.**
We can pass the `file name` (aside from the `avatar` which will be the blob URL) to the optimistic data when we upload the avatar, then pass the avatar file name to the `AttachmentModal` and use it as the default `file` state of the `AttachmentModal`. This will also work and quite straight-forward.

### What alternative solutions did you explore? (Optional)
Some alternative solutions for issue `2` mentioned in the original proposals [here](https://github.com/Expensify/App/issues/16528#issuecomment-1499949605).

**Comment by tienifr** [comment]
> This seems to be the actual root cause, I don't think there is a case where we'd want isLoading to be true when the component is first mounted

@eVoloshchak I think that proposal is suggesting to set `isLoading` to `true` when component is first mounted (it's currently `false` when first mounted).

Also it seems after applying the proposal the `height` and `width` are still `height: '100%', width: '100%'` initially (since `isLoading` is still true at the beginning) then changes to the actual small width, height of the image. So IMO that change might be hiding the bug somewhere rather than fixing it.

**Comment by hellohublot** [comment]
@eVoloshchak 

Thanks, I added some details to [my proposal](https://github.com/Expensify/App/issues/16528#issuecomment-1501105224), there are 3 bugs we need to solve in this issue

1. After the `AttachmentModal` is opened, a large picture `flashes`

2. When the user uses `AvatarCropModal to preview`, it is gif, but after uploading, it becomes `static png`. I assume we don’t need to support gif, so we need to display static pictures in AvatarCropModal as well

3. After the user's avatar is modified, the user quickly enters the DetailPage to quickly view his avatar, but the avatar cannot be loaded successfully because the `AttachmentModal.state.source` has not been updated, and the image.source loaded by the `AttachmentModal.props.source` but not `AttachmentModal.state.source`

**Comment by hellohublot** [comment]
@eVoloshchak 

Thanks for your suggestion, I added some ideas in [my proposal](https://github.com/Expensify/App/issues/16528#issuecomment-1501105224), this issue has 4 bugs in total, so I guess it can be divided into 4 issues, in order to get the best solution





> 3. After the user's avatar is modified, the user quickly enters the DetailPage to quickly view his avatar, but the avatar cannot be loaded successfully because the `AttachmentModal.state.source` has not been updated, I guess we still need to fix this bug, because the user opens the `DetailPage` on `one device`, but changes the `avatar` on `another device`, the `DetailPage.Avatar` will show the `new avatar`, but the `DetailsPage.AttachmentModal` will still show the `old avatar`
> 
> * we can update `AttachmentModal.state.source` in `AttachmentModal.componentDidUpdate` by `AttachmentModal.props.source`
> * We can also add a parameter `source` to `AttachmentModal.render.this.props.children.show`
> 
> 4. AttachmentView.Str.isImage cannot accurately judge blob://
> 
> * We can't be sure what type of file a blob:// or file:// is, maybe it's `not an image`, so I can't yet turn the `Str.isImage` condition into something like `Str.isImageUrl`, But we can use `displayFileInModal`, but we need to use `fetch` to convert it from `ObjectURL` to new `File`
> * Doesn't look like there is a good solution? so we can also try removing `DetailsPage.OfflineWithFeedback`, then replace in `PressableWithoutFocus.onPress(pendingAvatarAction ? null : show(avatarURL))`

**Comment by tienifr** [comment]
I’d say let’s focused on the 2 issues mentioned in the OP, the others if found during investigation, can be reported separately

**Comment by eVoloshchak** [comment]
There are a couple of good proposals, will review tomorrow as a first priority

**Comment by eVoloshchak** [comment]
Ok, I think we should focus on two bugs that are in the issue description. 
Those are two completely separate issues, but handling both of them in this issue seems fine.
> First Bug : Notice that the image fluctuates
Second Bug : Also, sometimes an empty attachment also gets displayed (This is not easily reproducible) For this, upload a large jpg/png file without cropping and try to preview the image immediately. It displays an empty attachment. - Not able to reproduce for me

**First Bug** 
@tienifr [solution](https://github.com/Expensify/App/issues/16528#issuecomment-1501676615) looks better to me, we're already using this approach, setting image dimensions only after it finished loading to prevent showing a preview with wrong dimensions. 
@hellohublot's solution also works, but we are still applying styles to the image that hasn't been loaded, which isn't right.

**Second bug**

@tienifr [solution](https://github.com/Expensify/App/issues/16528#issuecomment-1501676615) looks good to me too
> We can pass the file name (aside from the avatar which will be the blob URL) to the optimistic data when we upload the avatar, then pass the avatar file name to the AttachmentModal and use it as the default file state of the AttachmentModal. 

It ensures we can see the image right away (instead of waiting for it to be uploaded to the back end and loaded on the client) and uses an approach already used throughout the app

🎀👀🎀 C+ reviewed!
cc: @danieldoglas

**Comment by MelvinBot** [comment]
📣 @tienifr You have been assigned to this job by @danieldoglas!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by stephanieelliott** [comment]
PR is under review!

**Comment by danieldoglas** [comment]
Manually increasing the bounty since we found [another bug](https://expensify.slack.com/archives/C049HHMV9SM/p1682425619771099) while testing that will be fixed together [in this PR](https://github.com/Expensify/App/pull/17454).

**Comment by stephanieelliott** [comment]
PR is on staging

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.7-3 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17454


If no regressions arise, payment will be issued on 2023-05-05. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@eVoloshchak] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@eVoloshchak] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@eVoloshchak] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@eVoloshchak] Determine if we should create a regression test for this bug.
- [ ] [@eVoloshchak] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@stephanieelliott] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by danieldoglas** [comment]
@eVoloshchak can you please fill the checklist above?

@stephanieelliott I think this is ready for payment

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/8483

- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/8483#issuecomment-1540632149

- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: This is an edge case since it requires you to preview the image immediately after uploading, so it's hard to reproduce unless on a slow connection. I've asked here whether we should add "I throttled my network connection and tested it with slow connection to ensure it matches the expected behaviour" item to checklist

- Determine if we should create a regression test for this bug.
Is it easy to test for this bug? Yes
Is the bug related to an important user flow? (For example, adding a bank account) No
Is it an impactful bug? No

    I don't think regression test is needed here. This is reproducible using only a slow connection and has a very low impact

**Comment by stephanieelliott** [comment]
Getting ready to pay this - looks like the job was assigned on 4/13, all reviewed and all changes resolved on 4/18. Then on 4/18 another bug surfaced, we discussed and eventually decided to fix it in the same PR on 4/21. PR was finally merged on 4/16. 

For the purposes of calculating the speed bonus/penalty, I am going to use the date that comments were resolved on the original bug (4/18) as the "merge date". 4/13 assignment vs 4/18 merge = 5 days, so this one is not eligible for the speed bonus.
